### PR TITLE
Scheduler improvements #8786

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Review by anatol-sialitski — APPROVED** [review]

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java [hunk 1]
```diff
@@ -93,26 +91,32 @@ public void run()
...
     private Context taskContext()
     {
-        Context taskContext = ContextAccessor.current();
-
         if ( job.getUser() == null )
         {
-            return taskContext;
+            return ContextBuilder.from( ContextAccessor.current() ).
+                authInfo( AuthenticationInfo.create().
+                    principals( RoleKeys.AUTHENTICATED, RoleKeys.EVERYONE ).
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java:98]: Why would it be authenticated?

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java [hunk 2]
```diff
@@ -93,26 +91,32 @@ public void run()
...
+
+        final AuthenticationInfo authInfo = OsgiSupport.withService( SecurityService.class, securityService -> {
+            final VerifiedUsernameAuthToken token = new VerifiedUsernameAuthToken();
+            token.setIdProvider( job.getUser().getIdProviderKey() );
+            token.setUsername( job.getUser().getId() );
+
+            return securityService.authenticate( token );
+        } );
+
+        if ( !authInfo.isAuthenticated() )
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java:112]: Check only if user is specified

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java (hunk 2 above)
Comment (vbradnitski) [reply modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java:112]: There is a check "if ( job.getUser() == null )" above

Comment (rymsha) [reply modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java:108]: second thought: this is not Scheduler responsibility to verify identity, remove this check altogether (but keep `securityService.authenticate( token );`

**Review by rymsha — APPROVED** [review]

**Pull request merged**